nautobot/extras/jobs.py

```diff
@@ -965,7 +965,14 @@ def run_job(data, request, job_result_pk, commit=True, *args, **kwargs):
     """
     from nautobot.extras.models import JobResult  # avoid circular import
 
-    job_result = JobResult.objects.get(pk=job_result_pk)
+    # Getting the correct job result can fail if the stored data cannot be serialized.
+    # Catching `TypeError: the JSON object must be str, bytes or bytearray, not int`
```

Comment:
I don't quite get how you'd get this particular TypeError at this point in the code - perhaps adding an appropriate unit test case would make this clearer?

Reply:
Still an open question from my end.

Reply:
Previously it looks like when there is data that cant be serialised, it causes a type error where it cant get the object from the DB

Reply:
OK, so this'd be a case where `job_result_pk` is something other than a UUID? Can you test and verify that?

Reply:
No I think its more when the data stored in the DB is causing the issue

Reply:
@carbonarok Can you add a test case for the case where `JobResult.objects.get(pk=job_result_pk)` throws a TypeError exception?

Reply:
I havent been able to reproduce the error. It was reported in the same issue in Github, but no one can reproduce. I just added the error handling so it fails gracefully